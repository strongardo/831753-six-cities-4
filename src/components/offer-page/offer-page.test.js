import React from "react";
import renderer from "react-test-renderer";
import OfferPage from "./offer-page.jsx";

const offer = {
  name: `Beautiful & luxurious apartment at great location`,
  descriptions: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
  `An independent House, strategically located between Rembrand Square and National Opera,
  but where the bustle of the city comes to rest in this alley flowery and colorful.`],
  type: `Apartment`,
  owner: {
    url: `img/avatar-angelina.jpg`,
    name: `Angelina`,
    isSuper: true,
  },
  advantages: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`,
  ],
  price: 128,
  url: `img/apartment-01.jpg`,
  urls: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`],
  starsCount: 4,
  bedroomsCount: 4,
  guestsCount: 8,
  isPremium: true,
  id: 1,
};

it(`Should OfferPage render correctly`, () => {
  const tree = renderer
    .create(<OfferPage
      offer={offer}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
