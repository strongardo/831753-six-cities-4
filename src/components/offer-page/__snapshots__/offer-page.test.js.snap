// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferPage render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="/img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  mail@ln.kg
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The house among olive
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.2
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              6
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              813
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="/img/avatar-angelina.jpg"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                The house among olive
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <form
              className="reviews__form form"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <p
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="0-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="0-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={2}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={3}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={4}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={5}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                disabled={false}
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set
                   
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least
                   
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          id="map"
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="cities__place-card place-card"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                className="place-card__link"
                href="/offer/1"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    813
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  className="place-card__link"
                  href="/offer/1"
                  onClick={[Function]}
                >
                  The house among olive
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
